Sources/Service/WalletAvailabilityService.swift

```diff
@@ -2,22 +2,33 @@ import Foundation
 import Networking
 import UIKit
 
-protocol FeatureAvailabilityService {
+protocol FeatureAvailabilityService: AnyObject {
+    var hasAccessedBefore: Bool { get set }
     var shouldShowFeature: Bool { get }
-    func hasAccessedPreviously()
 }
 
 protocol UniversalLinkFeatureAvailabilityService {
     var shouldShowFeatureOnUniversalLink: Bool { get }
 }
 
-typealias WalletFeatureAvailabilityService = FeatureAvailabilityService & UniversalLinkFeatureAvailabilityService
+typealias WalletFeatureAvailabilityService = FeatureAvailabilityService & UniversalLinkFeatureAvailabilityService & SessionBoundData
 
 class WalletAvailabilityService: WalletFeatureAvailabilityService {
```

Comment:
Super pernickety but perhaps we should mark this as `final`, won't hold this up from being merged though 🙂

Reply:
Yes please, always `final class` 😄

Reply:
Done, thank you 🙂